### ISSUE TITLE
Task: Add Support for Mac & Windows Backends

### ISSUE THREAD

**Issue by Deim0s13** [Deim0s13/llm-assistant#19]
## 🛠 Task Summary

Update the model initialisation logic to support platform-specific execution backends: `mps` (Apple Silicon), `cuda` (NVIDIA), and `cpu`. This ensures compatibility when switching between macOS and Windows development environments.

---

### 🎯 Goal

Improve portability by enabling automatic device selection based on the available hardware, allowing seamless development across macOS (M1) and Windows (with NVIDIA GPU).

---

### 📂 Context

- Linked Epic: #6 
- Milestone: [v0.4.2](https://github.com/Deim0s13/llm-assistant/milestone/11)

This feature supports the ability to move between devices without changing any application code. It also sets the foundation for performance testing and future training/fine-tuning workflows.

---

### 🔧 Implementation Notes

- Update `initialize_model()` to set `device` using the following fallback order:
  ```python
  device = (
      "cuda" if torch.cuda.is_available()
      else "mps" if torch.backends.mps.is_available()
      else "cpu"
  )
- Log the selected device at startup
- Ensure the model.to(device) step is using the correct variable

---

### Acceptance Criteria

- [ ] Code selects cuda on Windows if available
- [ ] Code selects mps on macOS if available
- [ ] Falls back to cpu when neither is present
- [ ] Startup log shows selected device for visibility
- [ ] No changes required to use the project on either machine

---

### Labels

`feature`, `platform`, `enhancement`

---

### Status

Assign to milestone `v0.4.2` and track in the `Context Chaining & Memory Handling` epic.
